### PR TITLE
add frontend-test-utils with createExtensionTester

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

Work towards #19545, adding a utility for testing extensions in isolation. A new `frontend-test-utils` package is added to for the new `frontend-*-api` packages, mirroring `backend-test-utils`.

To simplify tests and properly forward errors this also adds `createSpecializedApp` to `frontend-app-api` as a synchronous version of `createApp`, which in turn is used by the new `createExtensionTester`.

Open question is whether the `.render()` method should be async or not. It's not necessary at this point but might be something we need in the future.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/frontend-test-utils/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+_This package was created through the Backstage CLI_.
+
+## Installation
+
+Install the package via Yarn:
+
+```sh
+cd <package-dir> # if within a monorepo
+yarn add @backstage/frontend-test-utils
```
Comment (camilaibs) [inline packages/frontend-test-utils/README.md:11]: Should it be a development dependency?

**Review by freben — APPROVED** [review]

File: packages/frontend-test-utils/package.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  },
+  "devDependencies": {
+    "@backstage/cli": "workspace:^",
+    "@testing-library/jest-dom": "^6.0.0"
+  },
+  "files": [
+    "dist"
+  ],
+  "dependencies": {
+    "@backstage/frontend-app-api": "workspace:^",
```
Comment (camilaibs) [inline packages/frontend-test-utils/package.json:33]: Just double-checking, it is fine to have a dependency on the app package here,right?

**Review by camilaibs — APPROVED** [review]

**Review by camilaibs — APPROVED** [review]
💯

**Review by camilaibs — APPROVED** [review]
💯

File: packages/frontend-test-utils/package.json (hunk 2 above)
Comment (Rugvip) [reply packages/frontend-test-utils/package.json:33]: Good check 😁 👍, but yep, it's on the users of this package to make sure it's a dev dep though

**Pull request merged**